### PR TITLE
allow nav poly to be directly edited via the vertices. This addresses #25459

### PR DESCRIPTION
**Pull request by MichaelJBrune**
Addressing #25459: this should now expose the vertices. Update the outline and poly to the new vertices.

Known Issue:

I found that modifying the outline in the editor doesn't change the vertices value unless you reselect the node. This is equal to the behavior in OccluderPolygon2D and CollisionPolygon2D.

### CODE REVIEW

**Comment by Zireael07** [comment]
I think you mean nav poly, not nav mesh? Nav mesh is 3D

File: scene/2d/navigation_polygon.cpp [hunk 1]
```diff
@@ -312,7 +315,7 @@ void NavigationPolygon::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_set_outlines", "outlines"), &NavigationPolygon::_set_outlines);
 	ClassDB::bind_method(D_METHOD("_get_outlines"), &NavigationPolygon::_get_outlines);
 
-	ADD_PROPERTY(PropertyInfo(Variant::POOL_VECTOR3_ARRAY, "vertices", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NOEDITOR | PROPERTY_USAGE_INTERNAL), "set_vertices", "get_vertices");
+	ADD_PROPERTY(PropertyInfo(Variant::POOL_VECTOR2_ARRAY, "vertices"), "set_vertices", "get_vertices");
```
Comment (akien-mga) [inline scene/2d/navigation_polygon.cpp]: The change to `POOL_VECTOR2_ARRAY` is a bugfix and could be merged right away, but the other changes need review from @reduz or anyone familiar with the notification system.

Comment (akien-mga) [reply scene/2d/navigation_polygon.cpp]: If this is approved, the same should likely be done in 3D for `NavigationMesh`.

Comment (akien-mga) [reply scene/2d/navigation_polygon.cpp]: I added the `POOL_VECTOR3_ARRAY` bugfix in #29285, so the rest should be rebased on top of the current master branch.

**Comment by MJBrune** [comment]
Erm, tried to rebase, broke everything. Still working on resolving it. 😊

**Comment by akien-mga** [comment]
I'll close for now as it's not in a mergeable state, and it might be simpler to do a new PR based on the current master branch (thus also giving the proposal more visibility than this old PR can provide).

**Pull request closed without merging**